src/OrleansAWSUtils/Membership/DynamoDBGatewayListProvider.cs

```diff
@@ -17,23 +19,24 @@ internal class DynamoDBGatewayListProvider : IGatewayListProvider
         private const string TABLE_NAME_DEFAULT_VALUE = "OrleansSiloInstances";
 
         private DynamoDBStorage storage;
-        private TimeSpan gatewayListRefreshPeriod;
         private string deploymentId;
         private readonly string INSTANCE_STATUS_ACTIVE = ((int)SiloStatus.Active).ToString();
         private readonly ILoggerFactory loggerFactory;
-        public DynamoDBGatewayListProvider(ILoggerFactory loggerFactory)
+        private readonly DynamoDBGatewayProviderOptions options;
+        private readonly TimeSpan maxStaleness;
+        public DynamoDBGatewayListProvider(ILoggerFactory loggerFactory, ClientConfiguration clientConfiguration, IOptions<DynamoDBGatewayProviderOptions> options)
         {
             this.loggerFactory = loggerFactory;
+            this.options = options.Value;
+            this.deploymentId = clientConfiguration.DeploymentId;
+            this.maxStaleness = clientConfiguration.GatewayListRefreshPeriod;
```

Comment:
This property seems like it should be in all gateway provider Option classes where it applies, doesn't it? Why wasn't it migrated from ClientConfiguration yet? or are there other uses for it?

Reply:
which property? GatwayListRefreshPeriod? 
GatewayListRefreshPeriod should be part of client config since client is the one which refreshes gateway, and client is where this parameter is used mainly. 

MaxStaleness isn't equal to GatewayRefreshPeriod in my understanding of IGatewayListProvider interface. The value of maxStaleness is specific to each gatewayListProvider's implementation.

Reply:
It is just happen to be the case that maxStaleness are all set to GatewatListRefreshPeriod in all our gatewayListProvider

Reply:
I see. Well, can't dig into it, but I'll trust your judgement, we can always move it later if we need, but you might be right that we can move it to a different Option class that is not gateway impl-specific

Reply:
I'm pretty sure it is the client refresh gateways through one of its component GatewayManager. 
Hence this setting shouldn't be part of GatewayListProviderOptions. GatewayListProvider doesn't refresh itself. 
As to in which way ClientConfiguration will be split to how many options, that is unknown to me. But I think this setting will belong to one of those options.